docs/dependencies.md

```diff
@@ -0,0 +1,72 @@
+
+# How to deal with dependencies
+
+For work with dependencies in Leapp repositories, you need to understand the
+dependencies types and why Leapp has a special way to deal with dependencies.
```

Comment:
I would reword this as `dependencies types` is something what people will think: I understand RPM (or in general: package) dependencies. Something like:
> problem we have with dependencies itself. Read this....